### ISSUE TITLE
Big issue using SCNIC 

### ISSUE THREAD

**Issue by vanesotjim** [lozuponelab/SCNIC#19]
**Hi, 
I am trying to run SCNIC,
First I ran, the script similar to the example, I just changed  the example with my table.biom, but I got a error**
SCNIC_analysis.py within -i table_filt_2_20Q_w_tax__VSEARCH_oref_dnvo_NM_NC.biom -o within_output/ -m sparcc
/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/biom/table.py:4064: FutureWarning: The SparseDataFrame class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version
  constructor = partial(pd.SparseDataFrame,
Traceback (most recent call last):
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 81, in <module>
    main()
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 77, in main
    args.func(**args_dict)
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/SCNIC/within_correls.py", line 68, in within_correls
    correls = ca.fastspar_correlation(table_filt, verbose=verbose, nprocs=procs)
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/SCNIC/correlation_analysis.py", line 77, in fastspar_correlation
    table.to_dataframe().to_dense().to_csv(path.join(temp, 'otu_table.tsv'), sep='\t', index_label='#OTU ID')
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/biom/table.py", line 4068, in to_dataframe
    return constructor(mat, index=index, columns=columns)

**Then I ran only the command, and again I got an error.**
SCNIC_analysis.py
Traceback (most recent call last):
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 81, in <module>
    main()
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 77, in main
    args.func(**args_dict)

I don't know what is the problem here. Please if you have some advice for me, this will be well received.
Thanks in advance.
Vanessa

**Comment by kthurimella** [comment]
@vanesotjim have you tried running the command $conda install -c bioconda -c conda-forge fastspar

This will ensure that fastspar is installed in your directory.

**Comment by marcjermaine-pontiveros** [comment]
Hi @vanesotjim, this is a bit late but it seems my pull request solves this issue. I got the same error when I'm running SCNIC for analysis. 

https://github.com/shafferm/SCNIC/pull/21

Line 77 from SCNIC/correlation_analysis.py causes this error:
```
table.to_dataframe().to_dense().to_csv(path.join(temp, 'otu_table.tsv'), sep='\t', index_label='#OTU ID')
```
I've replaced the line with the following:

```
# To fix AttributeError: 'DataFrame' object has no attribute 'to_dense'. 
# See: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sparse.to_dense.html
        table.to_dataframe().sparse.to_dense().to_csv(path.join(temp, 'otu_table.tsv'), sep='\t', index_label='#OTU ID')
```

**Comment by vanesotjim** [comment]
@kthurimella thank you for the advice, I already  did, but again, when I tried to run SCNIC y got the new error

**This is my command**
SCNIC_analysis.py within -i T_filt_2_20Q_w_tax__VSEARCH_oref_dnvo_NM_NC_hdf5.biom -o within_output/ -m sparcc
**I got this**
/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/biom/table.py:4064: FutureWarning: The SparseDataFrame class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version
  constructor = partial(pd.SparseDataFrame,
Traceback (most recent call last):
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 81, in <module>
    main()
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 77, in main
    args.func(**args_dict)
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/SCNIC/within_correls.py", line 68, in within_correls
    correls = ca.fastspar_correlation(table_filt, verbose=verbose, nprocs=procs)
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/SCNIC/correlation_analysis.py", line 77, in fastspar_correlation
    table.to_dataframe().to_dense().to_csv(path.join(temp, 'otu_table.tsv'), sep='\t', index_label='#OTU ID')
  File "/vault2/soft/miniconda2/envs/SCNIC/lib/python3.8/site-packages/biom/table.py", line 4068, in to_dataframe
    return constructor(mat, index=index, columns=columns)
**TypeError: SparseDataFrame() takes no arguments**

**And when I use only the command**  SCNIC_analysis.py
**I got** 
Traceback (most recent call last):
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 81, in <module>
    main()
  File "/vault2/soft/miniconda2/envs/SCNIC/bin/SCNIC_analysis.py", line 77, in main
    args.func(**args_dict)
**AttributeError: 'Namespace' object has no attribute 'func'**

**I am following the example usage from here** SCNIC: https://github.com/shafferm/SCNIC
**Sorry but I don't know how to solve this :(**

@marcjermaine-pontiveros thank you for the advice, I am not sure if the new issue that I got can solve with your recommendation it seems a little different? the procedure that you did in c5b76d8 have a good result?

Thanks in advance for your advice.

**Comment by anugos** [comment]
Hi, I am trying to run SCNIC with the example dataset given test folder but I am running through following error:
-------------------------------------------------------------------------------------------------------------------------------------
(base) [anugos24@c0166 scripts]$ SCNIC_analysis.py within -i fake_data.biom -o within_output/ -m sparcc
Traceback (most recent call last):
  File "/home/anugos24/miniconda3/bin/SCNIC_analysis.py", line 4, in <module>
    __import__('pkg_resources').run_script('SCNIC==0.6.3', 'SCNIC_analysis.py')
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/pkg_resources/__init__.py", line 662, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1466, in run_script
    exec(script_code, namespace, namespace)
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/SCNIC-0.6.3-py3.9.egg/EGG-INFO/scripts/SCNIC_analysis.py", line 4, in <module>
    __import__('pkg_resources').run_script('SCNIC==0.6.3', 'SCNIC_analysis.py')
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/SCNIC-0.6.3-py3.9.egg/SCNIC/within_correls.py", line 8, in <module>
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/biom/__init__.py", line 51, in <module>
    from .table import Table
  File "/home/anugos24/miniconda3/lib/python3.9/site-packages/biom/table.py", line 195, in <module>
    from ._filter import _filter
  File "biom/_filter.pyx", line 1, in init biom._filter
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
-------------------------------------------------------------------------------------------------------------------------------------
Kindly help me through. 
Thank you
-Anu

**Comment by sterrettJD** [comment]
I believe these issues are addressed in `0.6.3` but please feel free to open a new issue if you are still facing these problems.
